modules/deployment/src/test/java/org/mule/runtime/module/deployment/internal/ClassloadingTroubleshootingTestCase.java

```diff
@@ -359,7 +360,7 @@ public String describeFailure() {
         });
   }
 
-  private List<String> toMessages(List<LoggingEvent> loggingEvents) {
+  private List<String> toMessages(ImmutableList<LoggingEvent> loggingEvents) {
```

Comment:
I don't see what we gain by doing this change. Yes, we are not going to modify the list passed as a parameter, but we don't **need** it to be immutable either.
By changing it like this we are forcing the callers to take care of wrapping their list if they don't have it already immutable, when in fact the current implementation will return a new list, so the original one will remain safe from modifications.
I'd say, let's keep it as abstract as possible.